packages/markdown/processor/plugin/section-anchor-id.js

```diff
@@ -0,0 +1,16 @@
+const flatMap = require('unist-util-flatmap');
+const kebabCase = require('lodash/kebabCase');
+
+// Adds an empty <div id='section-slug'> next to all headings.
+// This is for compatibility with how we used to do slugs
+function transformer(ast) {
+  return flatMap(ast, node => {
+    if (node.type === 'element' && node.properties && node.properties.id) {
+      const id = `section-${kebabCase(node.properties.id)}`;
+      return [node, { type: 'element', tagName: 'div', properties: { id } }];
```

Comment:
One of the issues with Dom's initial take on this is that when you used the backwards-compatible hash link, the page would scroll _past_ the actual header Ideally we'd move this in to the node's `children` array. Not sure, but something like this...

```js
  // etc.
  if (/* blah && blah */) {
    node.children.push({ type: 'element', tagName: 'div', properties: { id } });
  }
  return [node];
}
```

...might work?

Reply:
Ahh that's interesting. Let me test it out!

Reply:
<img width="446" alt="Screen Shot 2020-05-05 at 5 44 30 PM" src="https://user-images.githubusercontent.com/8877659/81129391-04fe5880-8ef9-11ea-9a4e-c39494410296.png">

Reply:
@rafegoldberg updated.